Added registry check to allow system admins to disable uploads

Erik3003 opened the pull request:
I added the ability for system admins to disable uploads system wide by setting a registry key.
This is especially useful in corporate enviroments for work with confidential files and has been requested before in issue #2861
The registry key is: "HKEY_LOCAL_MACHINE\SOFTWARE\ShareX" and an entry needs to be "UploadForbidden" with the value a of "true" and needs to be created manually if a user wishes to use this feature.

Also added a missing German translation for "ThisFeatureWillNotWorkWhenDisableUploadOptionIsEnabled".

Jaex commented:
I'm not sure is it good idea to check for registry each upload, wouldn't it be better idea to check it once at startup of the app? Especially because most people not gonna use this system option, therefore each upload it will be doing unnecessary checks for most users.

Jaex commented:
Also I have concerns about people gonna require `HKEY_CURRENT_USER\SOFTWARE\ShareX` path to be used instead, so they can have user specific control on it.

Erik3003 commented:
I changed it to only check the registry on startup and have settings from HKEY_CURRENT_USER overwrite HKEY_LOCAL_MACHINE. Furthermore, I changed the entry to "UploadForbidden", to be consistent with the new variable naming.

Jaex commented:
You can store global variable in Program.cs instead of ApplicationConfig.cs because that file is used for settings which is saved to file. So it is not safe to store readonly global variables there.

Jaex commented:
Thanks!

Jaex commented:
Please note I changed registry name back to "DisableUpload".

BoBiene commented:
@Jaex any ETA / plans for the next release containing this feature?

Jaex commented:
There is no ETA.

@Erik3003 I think "LocalMachine" should have priority over "CurrentUser". For example if system admin set HKLM key and if user have access to HKCU then user can override system admins flag. So only if HKLM key is not exists then ShareX should check for HKCU key. Currently it is opposite.

sylveon commented:
The intent was probably to use in an environment where registry editor access was disabled to users, but yeah better do it that way (since they could use third party to write to the registry still)
